### ISSUE TITLE
BayesSearchCV: fit_params not getting set for the fit function of the estimator

### ISSUE THREAD

**Issue by prithagupta** [scikit-optimize/scikit-optimize#1132]
fit_params dict, optional
     Parameters to pass to the fit method.
When I pass the fit_params to the init function of the Bayes search CV, these parameters are not getting set to be passed to the fit function of the estimator:

```
import logging
from keras import Input, Model
from keras import backend as K
from keras.callbacks import EarlyStopping
from keras.layers import Dense, Activation
from keras.regularizers import l2
from keras.utils import to_categorical
from sklearn.base import BaseEstimator, ClassifierMixin
from tensorflow import optimizers
from keras.engine.base_layer import Layer
from keras.layers import Dense, Activation, BatchNormalization


class NormalizedDense(Layer):
    """Stop training when a monitored quantity has stopped improving.
    # Arguments
    units: Positive integer, dimensionality of the output space.
    activation: Activation function to use
        (see [activations](../activations.md)).
        If you don't specify anything, no activation is applied
        (ie. "relu:).
    normalize_before_activation: True if normalize the inputs before applying the activation.
    False if activation is applied before Bach Normalization
    """

    def __init__(self, units, activation="relu", **kwd):
        self.dense = Dense(units, activation="linear", **kwd)
        self.activation = Activation(activation=activation)
        self.batchnorm = BatchNormalization()
        self.norm_layer = None

    def __call__(self, x):
        return self.batchnorm(self.activation(self.dense(x)))

    def get_weights(self):
        w_b = self.batchnorm.get_weights()
        w_d = self.dense.get_weights()
        return w_b, w_d

    def set_weights(self, weights):
        w_b, w_d = weights
        self.batchnorm.set_weights(w_b)
        self.dense.set_weights(w_d)

class MultiLayerPerceptron(BaseEstimator, ClassifierMixin):
    def __init__(self, input_dim, n_classes, n_hidden=10, n_units=100, batch_normalization=True, activation='relu',
                 loss_function='categorical_crossentropy', solver='adam', metrics=['accuracy'], optimizer='sgd',
                 l=1e-4, kernel_initializer="lecun_normal", learning_rate=0.001,
                 early_stopping=False, model_save_path='', **kwargs):
        self.logger = logging.getLogger(name=MultiLayerPerceptron.__name__)
        self.input_dim = input_dim
        self.n_classes = n_classes
        self.n_units = n_units
        self.n_hidden = n_hidden
        self.batch_normalization = batch_normalization
        self.activation = activation
        self.loss_function = loss_function
        self.optimizer = optimizers.get(optimizer)
        self._optimizer_config = self.optimizer.get_config()
        K.set_value(self.optimizer.lr, learning_rate)
        self.solver = solver
        self.metrics = metrics
        self.learning_rate = learning_rate
        self.early_stopping = early_stopping
        self.model_save_path = model_save_path
        self.l = l
        self.kernel_regularizer = l2(l=self.l)
        self.kernel_initializer = kernel_initializer
        self.kwargs = kwargs
        self.model, self.scoring_model = None, None

    def _construct_layers(self, **kwargs):
        self.output_node = Dense(
            1, activation="sigmoid", kernel_regularizer=self.kernel_regularizer
        )
        self.input = Input(shape=self.input_dim, dtype='float32')
        if self.batch_normalization:
            self.hidden_layers = [
                NormalizedDense(self.n_units, name="hidden_{}".format(x), **kwargs)
                for x in range(self.n_hidden)
            ]
        else:
            self.hidden_layers = [
                Dense(self.n_units, name="hidden_{}".format(x), **kwargs)
                for x in range(self.n_hidden)
            ]
        self.score_layer = Dense(self.n_classes, activation=None, kernel_regularizer=self.kernel_regularizer)
        self.output_node = Activation('softmax', name='predictions')
        assert len(self.hidden_layers) == self.n_hidden

    def _construct_model_(self):
        x = self.hidden_layers[0](self.input)
        for hidden in self.hidden_layers[1:]:
            x = hidden(x)
            # x = BatchNormalization()(x)
        scores = self.score_layer(x)
        output = self.output_node(scores)
        model = Model(inputs=self.input, outputs=output, name="mlp_baseline")
        scoring_model = Model(inputs=self.input, outputs=scores, name="mlp_baseline_scorer")
        # model.compile(loss='categorical_crossentropy', optimizer=optimizer, metrics=['accuracy'])
        model.compile(loss=self.loss_function, optimizer=self.optimizer, metrics=self.metrics)
        scoring_model.compile(loss=self.loss_function, optimizer=self.optimizer, metrics=self.metrics)
        return model, scoring_model

    def reshape_inputs(self, y):
        if y is not None:
            y = to_categorical(y, num_classes=self.n_classes)
        return y

    def fit(self, X, y, epochs=10, batch_size=10, callbacks=None, validation_split=0.1, verbose=0, **kwd):
        self._construct_layers(
            kernel_regularizer=self.kernel_regularizer,
            kernel_initializer=self.kernel_initializer,
            activation=self.activation,
            **self.kwargs
        )
        self.model, self.scoring_model = self._construct_model_()
        y = self.reshape_inputs(y)
        er = EarlyStopping(monitor="val_loss", patience=10, restore_best_weights=True)
        if self.early_stopping:
            if callbacks is not None:
                callbacks.append(er)
            else:
                callbacks = [er]
        self.model.fit(x=X, y=y, batch_size=batch_size, verbose=verbose,
                       validation_split=validation_split, epochs=epochs, callbacks=callbacks)

        return self

    def predict(self, X, verbose=0):
        predictions = self.model.predict(x=X, verbose=verbose)
        return predictions

    def score(self, X, y, sample_weight=None, verbose=0):
        y = self.reshape_inputs(y)
        model_metrics = self.model.evaluate(x=X, y=y, verbose=verbose)[-1]
        return model_metrics

    def predict_proba(self, X, verbose=0):
        prob_predictions = self.model.predict_proba(x=X, verbose=verbose)
        return prob_predictions

    def decision_function(self, X, verbose=0):
        prob_predictions = self.model.predict_proba(x=X, verbose=verbose)
        return prob_predictions

    def get_params(self, deep=True):
        out = dict()
        for key in self._get_param_names():
            value = getattr(self, key)
            if deep and hasattr(value, "get_params"):
                deep_items = value.get_params().items()
                out.update((key + "__" + k, val) for k, val in deep_items)
            out[key] = value
        return out

    def set_params(self, **parameters):
        for parameter, value in parameters.items():
            setattr(self, parameter, value)
        return self

```

```
from skopt import BayesSearchCV
from sklearn.datasets import load_iris
from skopt.space import Real, Categorical, Integer
from pycilt.multi_layer_perceptron import MultiLayerPerceptron
from sklearn.model_selection import train_test_split
import numpy as np
import logging


X, y = load_iris(return_X_y=True)
X_train, X_test, y_train, y_test = train_test_split(X, y,
                                                    train_size=0.75,
                                                    random_state=0)
input_dim = X_train.shape[-1]
n_classes = len(np.unique(y_train))
# log-uniform: understand as search over p = exp(x) by varying x
fit_params = {'epochs':50, 'batch_size':50 ,'verbose':1}
opt = BayesSearchCV(
    MultiLayerPerceptron(input_dim=input_dim, n_classes=n_classes),
    {
        'learning_rate': Real(1e-6, 1e-1, prior='log-uniform'),
        'n_hidden': Integer(1,4),
        'n_units': Integer(1,8),
    },
    n_iter=5,
    random_state=0,
    fit_params=fit_params
)
def callback(logger):
    def on_step(opt_result):
        """
        Callback meant to view scores after
        each iteration while performing Bayesian
        Optimization in Skopt"""
        points = opt_result.x_iters
        scores = -opt_result.func_vals
        logger.info('Next parameters: {}, accuracy {}'.format(points[-1], scores[-1]))
    return on_step
logger = logging.getLogger('Classifier-Test')
# executes bayesian optimization
_ = opt.fit(X_train, y_train, callback=callback(logger))
```
The output of this function is 
epochs 10 batch_size 10
epochs 10 batch_size 10
epochs 10 batch_size 10
epochs 10 batch_size 10

But when I pass the parameters to the fit function, then they are set properly

`_ = opt.fit(X_train, y_train, callback=callback(logger), **fit_params)`
epochs 50 batch_size 50
Epoch 1/50
2/2 [==============================] - 1s 186ms/step - loss: 1.9688 - accuracy: 0.1375 - val_loss: 1.2980 - val_accuracy: 0.4444
Epoch 2/50
2/2 [==============================] - 0s 27ms/step - loss: 1.9366 - accuracy: 0.1750 - val_loss: 1.2887 - val_accuracy: 0.4444
Epoch 3/50
2/2 [==============================] - 0s 25ms/step - loss: 1.9242 - accuracy: 0.1375 - val_loss: 1.2799 - val_accuracy: 0.4444
Epoch 4/50
2/2 [==============================] - 0s 23ms/step - loss: 1.9046 - accuracy: 0.1750 - val_loss: 1.2707 - val_accuracy: 0.4444
Epoch 5/50
2/2 [==============================] - 0s 23ms/step - loss: 1.9284 - accuracy: 0.1875 - val_loss: 1.2610 - val_accuracy: 0.4444
Epoch 6/50

I think the fit_params which are passed in the init function should be used as the parameters for the estimator not the one we pass to the fit function 

Versions for scikit-learn and scikit optimize:

scikit-learn              1.1.3                    pypi_0    pypi
scikit-optimize           0.9.0                    pypi_0    pypi
